src/main/scala/com/stratio/khermes/helpers/faker/generators/NameGenerator.scala

```diff
@@ -24,21 +24,45 @@ import com.typesafe.scalalogging.LazyLogging
 /**
  * Generates random names.
  */
-case class NameGenerator(locale: String) extends FakerGenerator
+case class NameGenerator(locale: String, strategy: Option[String]) extends FakerGenerator
   with AppSerializer
   with LazyLogging {
 
   override def name: String = "name"
 
-  lazy val nameModel = locale match {
+  lazy val nameModel: Seq[Either[String, NameModel]] = locale match {
     case AppConstants.DefaultLocale =>
       val resources = getResources(name)
-        .map(parse[NameModel](name, _))
-      if (parseErrors[NameModel](resources).nonEmpty) logger.warn(s"${parseErrors[NameModel](resources)}")
-      resources
-    case localeValue => Seq(parse[NameModel](name, s"$localeValue.json"))
+      if (strategy.isDefined) {
+        val res = resources.map(parse[NameModel](name, _)
+          .right.map(elem =>
+          NameModel(
+            repeatElementsInList(listWithStrategy(elem.firstNames, strategy.getOrElse(throw new IllegalArgumentException("Bad strategy definition"))))
+            , repeatElementsInList(listWithStrategy(elem.lastNames, strategy.getOrElse(throw new IllegalArgumentException("Bad strategy definition")))))
+        ))
+        if (parseErrors[NameModel](res).nonEmpty) logger.warn(s"${parseErrors[NameModel](res)}")
+        res
+      }
```

Comment:
put:
} else {

not
}
else {

Review your PR because i think that you have more ot this and it is not following our coding starndar.